app/controllers/payments/WhichVatPeriodToPayController.scala

```diff
@@ -0,0 +1,101 @@
+import pages.{JourneyRecoveryPage, Waypoints, YourAccountPage}
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import play.api.{Configuration, Logging}
+import services.PaymentsService
+import uk.gov.hmrc.domain.Vrn
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.payments.{NoPaymentsView, WhichVatPeriodToPayView}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class WhichVatPeriodToPayController @Inject()(
+                                               cc: AuthenticatedControllerComponents,
+                                               config: Configuration,
+                                               paymentsService: PaymentsService,
+                                               formProvider: WhichVatPeriodToPayFormProvider,
+                                               view: WhichVatPeriodToPayView,
+                                               viewNoPayment: NoPaymentsView
+                                             )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport with Logging {
+
+  private val form = formProvider()
+  protected val controllerComponents: MessagesControllerComponents = cc
+
+  val paymentsBaseUrl = config.get[Service]("microservice.services.pay-api")
+
+  def onPageLoad(waypoints: Waypoints): Action[AnyContent] = cc.authAndRequireData().async {
+    implicit request => {
+      val prepareFinancialData: Future[PrepareData] = paymentsService.prepareFinancialData()
+      prepareFinancialData.flatMap { pfd =>
+        val payments = pfd.duePayments ++ pfd.overduePayments
+        val paymentError = payments.exists(_.paymentStatus == PaymentStatus.Unknown)
+
+        if (payments.isEmpty) {
+          Future.successful(Ok(viewNoPayment(YourAccountPage.route(waypoints).url)))
+        } else if (payments.size == 1) {
+          makePayment(request.vrn, payments.head)
+        } else {
+          Future.successful(Ok(view(form, payments, paymentError = paymentError)))
+        }
+      }
+    }
+  }
+
+  private def makePayment(vrn: Vrn, payment: Payment)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Result] = {
+    paymentsService.makePayment(vrn, payment.period, payment).map {
+      case Right(value) => Redirect(value.nextUrl)
+      case _ => Redirect(s"$paymentsBaseUrl/pay/service-unavailable")
+    }
+  }
+
+  def onSubmit(waypoints: Waypoints): Action[AnyContent] = cc.authAndRequireData.async {
+    implicit request => {
+      val prepareFinancialData: Future[PrepareData] = paymentsService.prepareFinancialData()
+      prepareFinancialData.flatMap { pfd =>
+        val payments = pfd.duePayments ++ pfd.overduePayments
+        val paymentError = payments.exists(_.paymentStatus == PaymentStatus.Unknown)
+        if (payments.size == 1) {
+          makePayment(request.vrn, payments.head)
```

Comment:
```
        payments match {
          case Seq(payment) => makePayment(request.vrn, payment)
          case _ => ...
        }
```

Reply:
100%. done.

Reply:
Did you push the commit? I don't see the change